WooCommerce/Classes/ViewRelated/Orders/OrdersViewController.swift

```diff
@@ -422,7 +422,7 @@ private extension OrdersViewController {
             return
         }
 
-        if currentSiteStatuses.contains(statusFilter) == false {
+        if !currentSiteStatuses.contains(where: { $0.name == statusFilter.name && $0.slug == statusFilter.slug }) {
```

Comment:
🤔 my first instinct would be to encapsulate that filter into the object that models the status, but then, I was thinking... isn't that a match for equality? I mean, if the name and slug are the same, the statuses are the same, right? So, in theory, if we made the OrderStatus equatable, we could go back to just check for `contains`

Also, and this is a personal preference, and therefore a highly subjective comment, but I find if clauses that start with a negation (`!`) really difficult to read. But that's just me

Reply:
> my first instinct would be to encapsulate that filter into the object that models the status, but then, I was thinking... isn't that a match for equality? I mean, if the name and slug are the same, the statuses are the same, right? So, in theory, if we made the OrderStatus equatable, we could go back to just check for contains. Or am I missing something?

My first change was also to update the `Equatable` implementation, but I felt that equality is generally defined to include the variable fields and the exclusion of `count` is specific to this use case.

> Also, and this is a personal preference, and therefore a highly subjective comment, but I find if clauses that start with a negation (!) really difficult to read. But that's just me

I slightly prefer `!` because I can just tell whether it's true or not from the beginning 😄I see in the code we have a mix `!` and `== false`, maybe this is something we can pick one as a team if some of us feel more strongly about it?

Reply:
I have no preference, but will follow whichever convention everyone else wants.